### DISCUSSION TITLE
Simple Clocks and Advanced Clocks

### DISCUSSION THREAD

**Discussion by OmniacDev** [SebLague/Digital-Logic-Sim#92] [Show and tell]
Made some clock circuits I thought I'd post here:

### Simple Clock

Building this clock is a bit tedious because you need to build it in an exact order, or else it doesn't work.

(Sorry about the video quality in advance, I recorded this on my crappy laptop) 

**STEP 1:** First you want to make an "equals" circuit. This is simply here so that we are able to loop the "not" circuit back into itself. You can see how to make this in the video below:

https://user-images.githubusercontent.com/64994204/156973351-456fec59-4ca4-41c7-9135-e56521993ee2.mp4

**STEP 2:** Now we can start making the actual clock. This clock has to be built in an exact order to work (There's some kind of bug that only lets the clock work if it is made in a certain order). 

The order goes like this:

1. Place an input pin called "IN".
2. Place an output pin called "OUT".
3. Drop in an "AND" circuit.
4. Drop in a "=" or "EQUALS" circuit that we made earlier.
5. Finally, Drop in a "NOT" circuit.

After following this exact order, you can start wiring up the circuits as shown in the video below:

https://user-images.githubusercontent.com/64994204/156973788-8d25c0fe-30e8-48ca-aef4-ef23a8ccd7e1.mp4

**STEP 3:** Now you can package it up into a "CLOCK" or "CLK" circuit, and if you've follow the steps correctly, you should see the output pin oscillating from on to off. Unfortunately, the clock only runs at one speed, as I haven't figured out how to slow it yet.

### Advanced Clock

Now, using the simple clock we made earlier, we can make a more advanced clock that includes an "ON" pin, a "OFF" / "KILL" pin, and a "STEP" pin (To step the clock forward once at a time).

**STEP 1:** This clock is a lot less tedious to build, and doesn't require you to build it in an exact order. To build the advanced clock, please follow the steps in the video below:

https://user-images.githubusercontent.com/64994204/156974297-e90dfe83-d43a-48ad-8bfa-2be31e079eed.mp4

Thanks for reading this, I hope it helped satisfy your need for a clock circuit.

**Comment by CallumCM** [comment]
Woahh! This is so cool! Thank you!

**Reply by OmniacDev** [reply]
No problem :)
